dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -837,22 +838,20 @@ prettyCharacterSet characterSet expression =
         prettyApplicationExpression a0
 
     prettyApplicationExpression :: Pretty a => Expr Src a -> Doc Ann
-    prettyApplicationExpression a0 = case a0 of
-        App   {} -> result
-        Some  {} -> result
-        ToMap {} -> result
-        Merge {} -> result
-        Note _ b -> prettyApplicationExpression b
-        _        -> prettyImportExpression a0
+    prettyApplicationExpression = \case
+        App a b           -> f (prettyApplicationExpression a) [b]
+        Some a            -> f (builtin "Some") [a]
+        ToMap a Nothing   -> f (keyword "toMap") [a]
+        Merge a b Nothing -> f (keyword "merge") [a, b]
+        Note _ b          -> prettyApplicationExpression b
+        e                 -> prettyImportExpression e
       where
-        result = enclose' "" "" " " "" (reverse (docs a0))
-
-        docs (App  a b) = ( prettyImportExpression b, Pretty.indent 2 (prettyImportExpression b) ) : docs a
```

Comment:
My refactoring broke the formatting of application-expressions where the first arguments are short:

```
f x
  rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr
```

should be

```
f
  x
  rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr
```

I'll try again

Reply:
What I don't understand is why record fields are still formatted correctly:

```
{ a =
    f
      x
      rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr
}
```